File: ios/Classes/Ads/LingYeSplashHandler.swift
import Foundation
import Flutter
import CJMobileAd

class LingYeSplashHandler: NSObject {

    static let shared: LingYeSplashHandler = LingYeSplashHandler()
    
    var model: BottomModel?
    
    var splashAd: CJSplashAd?

    func handleMethodCall(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        model = BottomModel(adConfig: call.arguments as? [String : Any] ?? [:])
        if call.method == LingYeAdSdkMethodNames.loadAndShowSplashAd {
            loadAndShowSplashAd()
            result(true)
        }
        result(nil)
    }

    func loadAndShowSplashAd() {
        LingYeTools.shared.lyLog(msg: "开始调用开屏")
        DispatchQueue.main.async {
            let bottomView = LingYeBottomView.init(bottomModel: self.model)
            self.splashAd = CJSplashAd(resourceId: self.model?.adId as? String ?? "", rootViewController: LingYeTools.shared.currentViewController() ?? UIViewController(), customLogoView: bottomView)
            self.splashAd?.delegate = self
            self.splashAd?.loadData()
        }
    }
    
}

extension LingYeSplashHandler: CJSplashAdDelegate {
    
    func splashAdDidLoad(_ splashAd: CJSplashAd, resourceId: String) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.splashAdLoadSuccess, arguments: ["code": 0, "message": "开屏加载成功"])
        splashAd.show(to: LingYeTools.shared.lyGetCurrentWindow() ?? UIWindow())
    }
    
    func splashAdLoadFailed(_ splashAd: CJSplashAd, error: any Error) {
        let msg = error.localizedDescription
        let code = -1
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.splashAdLoadFailed, arguments: ["code": code, "message": msg])
    }
    
    func splashAd(onShow splashAd: CJSplashAd) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.splashAdOnShow, arguments: ["code": 0, "message": "开屏展示成功"])
    }
    
    func splashAd(onClicked splashAd: CJSplashAd) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.splashAdOnClick, arguments: ["code": 0, "message": "开屏触发点击"])
    }
    
    func splashAd(onClosed splashAd: CJSplashAd) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.splashAdOnClose, arguments: ["code": 0, "message": "开屏触发关闭"])
    }
}

File: android/src/main/kotlin/com/example/flutter_cjadsdk_plugin/ads/LingYeSplashHandler.kt
import android.view.ViewGroup
import cj.mobile.CJSplash
import cj.mobile.listener.CJSplashListener
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class LingYeSplashHandler {

    var splashAd: CJSplash? = null
    var params: Map<String, Any>? = null

    companion object {
        @Volatile
        private var instance: LingYeSplashHandler? = null
        fun getInstance(): LingYeSplashHandler {
            return instance ?: synchronized(this) {
                instance ?: LingYeSplashHandler().also { instance = it }
            }
        }
    }

    fun handleMethodCall(call: MethodCall, result: MethodChannel.Result) {
        val method: String = call.method
        val flutterParams: Map<String, Any>? = call.arguments as? Map<String, Any>
        params = flutterParams
        when (method) {
            LingYeAdSdkMethodNames.loadAndShowSplashAd -> {
                loadAndShowSplashAd()
                result.success(true)
            }
            else -> result.notImplemented()
        }
    }

    private fun loadAndShowSplashAd() {
        splashAd = CJSplash()
        val activity = LingYeTools.getActivity()
        val adId = params?.get("adId")
        val map: MutableMap<String, Any> = HashMap()

        var content = activity?.findViewById<ViewGroup>(android.R.id.content)
        if (activity == null || content == null) {
            map["code"] = -1
            map["message"] = "activity或者content为空，请检查请求时机"
            LingYeBaseRegister.getInstance().sendMessageToFlutter(
                LingYeAdSdkCallBackMethodNames.splashAdLoadFailed,
                map
            )
            return
        }


        val handle = BottomViewHandle()
        val model = LingYeBottomModel()
        params?.let { model.setupAdConfig(it) }
        handle.setupBottom(model)

        splashAd?.loadAd(activity, adId.toString(), handle.getWidth(), handle.getHeight(), object : CJSplashListener {
            override fun onShow() {
                map.put("code", 0)
                map.put("message", "展示成功")
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.splashAdOnShow,
                    map
                )
            }

            override fun onError(s: String, s1: String) {
                map["code"] = -1
                map["message"] = s+s1
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.splashAdLoadFailed,
                    map
                )
            }

            override fun onLoad() {
                map.put("code", 0)
                map.put("message", "加载成功")
                (activity.window?.decorView as ViewGroup?)!!.removeView(handle.content)
                (activity.window?.decorView as? ViewGroup)!!.addView(handle.content)
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.splashAdLoadSuccess,
                    map
                )
                splashAd?.showAd(activity, handle.splashContent)
            }

            override fun onClick() {
                map.put("code", 0)
                map.put("message", "触发点击")
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.splashAdOnClick,
                    map
                )
            }

            override fun onClose() {
                map.put("code", 0)
                map.put("message", "触发关闭")
                (activity.window?.decorView as ViewGroup?)!!.removeView(handle.content)
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.splashAdOnClose,
                    map
                )
            }
        })
    }

}